v1/python-sdk/tutorials/automl-with-azureml/forecasting-demand-forecasting-many-models/auto-ml-forecasting-demand-forecasting-many-models.ipynb

```diff
@@ -0,0 +1,1626 @@
+    "    one_forecast.sort_values(time_column_name, inplace=True)\n",
+    "    _draw_one_plot(\n",
+    "        one_forecast,\n",
+    "        time_column_name,\n",
+    "        target_column_name,\n",
+    "        [GRAIN_COL],\n",
+    "        [target_column_name, \"predicted\", \"predicted_base\"],\n",
+    "        pdf,\n",
+    "        plot_predictions=True,\n",
+    "    )\n",
+    "pdf.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 7.3 Calculate metrics\n",
+    "Here, we will calculate the metric of interest for each day. For illustration purposes we use root mean squared error as the metric of choice. However, the `compute_all_metrics` method calculated all primary and secondary metrics for AutoML runs. Please refer to this <u>*Regression/forecasting metrics*</u> section in this [document](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-understand-automated-ml#regressionforecasting-metrics) for the list of available metrics. We will calculate the distribution of this metric for each time series in our dataset. Looking at the descrptive stats of such metrics can be more informative than calculating a single metric such as the mean for each time series. As an example, we are looking at the RMSE (root mean squared error) metric, but you can choose any other metric computed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scripts.helper_scripts import compute_all_metrics\n",
+    "\n",
+    "metrics_per_grain_day = compute_all_metrics(\n",
+    "    fcst_df=backtest,\n",
+    "    actual_col=target_column_name,\n",
+    "    fcst_col=\"predicted\",\n",
+    "    ts_id_colnames=[GRAIN_COL, PERIOD_COLUMN],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\n",
+    "    f'List of available metrics: {metrics_per_grain_day[\"metric_name\"].unique()}\\n---'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DESIRED_METRIC_NAME = \"root_mean_squared_error\"\n",
+    "\n",
+    "metrics_per_grain_day = metrics_per_grain_day.query(\n",
+    "    f'metric_name == \"{DESIRED_METRIC_NAME}\"'\n",
+    ")\n",
+    "metrics_per_grain_day[[GRAIN_COL, PERIOD_COLUMN]] = metrics_per_grain_day[\n",
+    "    \"time_series_id\"\n",
+    "].str.split(\"|\", 1, expand=True)\n",
+    "metrics_per_grain_day.to_csv(\n",
+    "    os.path.join(OUTPUT_DIR, \"metrics-automl.csv\"), index=False\n",
+    ")\n",
+    "metrics_per_grain_day.groupby(GRAIN_COL)[\"metric\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# baseline metrics\n",
```

Comment:
Let us move this comment to the markdown cell above the code.

Reply:
done.